src/asmcnc/job/job_data.py

```diff
@@ -405,3 +409,27 @@ def post_job_data_update_post_send(self):
         self.post_production_notes = ''
         self.batch_number = ''
         self.percent_thru_job = 0
+
+    def get_excluded_line_numbers(self, gcode):
+        self.g0_lines[:] = []
+        for i, line in enumerate(gcode):
+            if 'G0' in line:
+                self.g0_lines.append(i)
+                for j, next_line in enumerate(gcode[i + 1:]):
+                    if 'G1' in next_line or 'G2' in next_line or 'G3' in next_line:
+                        break
+                    else:
+                        self.g0_lines.append(i + j + 1)
+        return self.g0_lines
+
+    def get_spindle_speeds(self, gcode):
```

Comment:
when does this get called? Has potential to be slow and locking bc it's a for loop

Reply:
inside run_job in serial_connection